src/lib/SX1280Driver/SX1280_Regs.h

```diff
@@ -0,0 +1,258 @@
+#pragma once
+
+#define REG_LR_FIRMWARE_VERSION_MSB 0x0153 //The address of the register holding the firmware version MSB
+#define SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB 0x0954
+#define SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MASK 0x0FFFFF
+
+#define SX1280_XTAL_FREQ 52000000
+#define SX1280_FREQ_STEP ((double)(SX1280_XTAL_FREQ / pow(2.0, 18.0)))
+
+typedef enum
+{
+    SX1280_RF_IDLE = 0x00, //!< The radio is idle
+    SX1280_RF_RX_RUNNING,  //!< The radio is in reception state
+    SX1280_RF_TX_RUNNING,  //!< The radio is in transmission state
+    SX1280_RF_CAD,         //!< The radio is doing channel activity detection
+} SX1280_RadioStates_t;
+
+/*!
+ * \brief Represents the operating mode the radio is actually running
+ */
+typedef enum
+{
+    SX1280_MODE_SLEEP = 0x00, //! The radio is in sleep mode
+    SX1280_MODE_CALIBRATION,  //! The radio is in calibration mode
+    SX1280_MODE_STDBY_RC,     //! The radio is in standby mode with RC oscillator
+    SX1280_MODE_STDBY_XOSC,   //! The radio is in standby mode with XOSC oscillator
+    SX1280_MODE_FS,           //! The radio is in frequency synthesis mode
+    SX1280_MODE_RX,           //! The radio is in receive mode
+    SX1280_MODE_TX,           //! The radio is in transmit mode
+    SX1280_MODE_CAD           //! The radio is in channel activity detection mode
+} SX1280_RadioOperatingModes_t;
+
+#define SX1280_RX_TX_CONTINUOUS \
+    (TickTime_t) { RADIO_TICK_SIZE_0015_US, 0xFFFF }
+#define SX1280_RX_TX_SINGLE \
+    (TickTime_t) { RADIO_TICK_SIZE_0015_US, 0 }
+
+/*!
+ * \brief Declares the oscillator in use while in standby mode
+ *
+ * Using the STDBY_RC standby mode allow to reduce the energy consumption
+ * STDBY_XOSC should be used for time critical applications
+ */
+typedef enum
+{
+    SX1280_STDBY_RC = 0x00,
+    SX1280_STDBY_XOSC = 0x01,
+} SX1280_RadioStandbyModes_t;
+
+/*!
+ * \brief Declares the power regulation used to power the device
+ *
+ * This command allows the user to specify if DC-DC or LDO is used for power regulation.
+ * Using only LDO implies that the Rx or Tx current is doubled
+ */
+typedef enum
+{
+    SX1280_USE_LDO = 0x00,  //! Use LDO (default value)
+    SX1280_USE_DCDC = 0x01, //! Use DCDC
+} SX1280_RadioRegulatorModes_t;
+
+/*!
+ * \brief Represents the possible packet type (i.e. modem) used
+ */
+typedef enum
+{
+    SX1280_PACKET_TYPE_GFSK = 0x00,
+    SX1280_PACKET_TYPE_LORA,
+    SX1280_PACKET_TYPE_RANGING,
+    SX1280_PACKET_TYPE_FLRC,
+    SX1280_PACKET_TYPE_BLE,
+    SX1280_PACKET_TYPE_NONE = 0x0F,
+} SX1280_RadioPacketTypes_t;
+
+//only for GFSK
```

Comment:
Do we want this? This one might actually be useful to keep?

Reply:
Yeah lets keep